examples/Example7/example7-geant4.cc

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+#include "geant4_mock.h"
+
+#include <AdePT/ArgParser.h>
+
+#include <G4GDMLParser.hh>
+#include <G4LogicalVolumeStore.hh>
+#include <G4MaterialCutsCouple.hh>
+#include <G4ProductionCutsTable.hh>
+#include <Randomize.hh>
+
+#include <G4HepEmRunManager.hh>
+#include <G4HepEmCLHEPRandomEngine.hh>
+#include <G4HepEmData.hh>
+#include <G4HepEmState.hh>
+#include <G4HepEmParameters.hh>
+#include <G4HepEmMatCutData.hh>
+#include <G4HepEmDataJsonIO.hh>
+
+int main(int argc, char* argv[])
+{
+  // Only outputs are the data file(s)
+  // Separate for now, but will want to unify
+  OPTION_STRING(gdml_file, "example7.gdml");
+  std::string g4hepem_file = gdml_file + ".g4hepem";
+
+  // Build mock geant4 setup
+  // - Should create geometry, regions and cuts
+  G4PVPlacement* world = geant4_mock();
+
+  // Construct the G4HepEmRunManager, which will fill the data structures
+  // on calls to Initialize
+  auto* runMgr    = new G4HepEmRunManager(true);
+  auto* rngEngine = new G4HepEmCLHEPRandomEngine(G4Random::getTheEngine());
+  runMgr->Initialize(rngEngine, 0);
+  runMgr->Initialize(rngEngine, 1);
+  runMgr->Initialize(rngEngine, 2);
```

Comment:
Is there a particular reason to use `G4HepEmRunManager`? I discussed this with Mihaly initially, and AdePT should not depend on the `G4HepEm` library because we don't need their TLData and such. I'd prefer to initialize only the needed data structures directly, as we're doing in all other examples.

Reply:
No other reason than simplicity (and following what's in the G4HepEM tests/examples) - agree though it would be better to have direct initialization, but that's an addition for G4HepEm.

Reply:
No, it's already there and we're using that in AdePT, see https://github.com/apt-sim/AdePT/blob/0165af581767edcfc43afdcc4429369270de72ae/examples/TestEm3/TestEm3.cu#L51-L55

Reply:
I don't see the point of replacing 4 calls that you can call selectively by 1 call, but I'll let Mihaly decide

Reply:
The code still seems to be there...

Reply:
Indeed, @drbenmorgan you are not using here yet the interface introduced in mnovak42/g4hepem#47, can you update the code to merge this?

Reply:
Apologies, the change will need #121 merged to pick up the new interface from mnovak42/g4hepem#47